File: Android_App_Java_Gucci_Store/app/src/main/java/com/example/dhura_mistry_project2/CartAdapter.java
package com.example.dhura_mistry_project2;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.FirebaseFirestore;
import com.squareup.picasso.Picasso;

import java.util.List;

public class CartAdapter extends RecyclerView.Adapter<CartAdapter.MyViewHolder> {

//    private final ProductsDetailInterface productsDetailInterface;
    Context context;
    List<CartItems> list;
    int totalPrice = 0;
    FirebaseFirestore firestore;
    FirebaseAuth auth;

    public CartAdapter(Context context, List<CartItems> list) {
        this.context = context;
        this.list = list;
        firestore = FirebaseFirestore.getInstance();
        auth = FirebaseAuth.getInstance();
    }

    @NonNull
    @Override
    public CartAdapter.MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(context).inflate(R.layout.cart_cardview,parent,false);
        return new CartAdapter.MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull CartAdapter.MyViewHolder holder, @SuppressLint("RecyclerView") int position) {

        CartItems CarItems = list.get(position);
        holder.itemName.setText(CarItems.getItemName());
        holder.itemPrice.setText("$" +CarItems.getItemPrice());
        holder.itemQuantity.setText("Quantity: "+String.valueOf(CarItems.getQuantity()));
        Picasso.get().load(CarItems.getItemImage()).into(holder.itemImage);

        //pass total amount to cart
        totalPrice = totalPrice + CarItems.getTotalPrice();

        Intent intent = new Intent("MyTotalAmount");
        intent.putExtra("totalAmount",String.valueOf(totalPrice));
        LocalBroadcastManager.getInstance(context).sendBroadcast(intent);

        holder.btnRemove.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                firestore.collection("AddToCart").document(auth.getCurrentUser().getUid()).collection("CurrentUser").document(list.get(position).getDocumentId()).delete().addOnCompleteListener(new OnCompleteListener<Void>() {
                    @Override
                    public void onComplete(@NonNull Task<Void> task) {

                        if(task.isSuccessful()){
                            list.remove(list.get(position));
                            notifyDataSetChanged();
                            Toast.makeText(context,"Item deleted",Toast.LENGTH_SHORT).show();
                        }else{
                            Toast.makeText(context,"Error"+task.getException(),Toast.LENGTH_SHORT).show();
                        }
                    }
                });

            }
        });

    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder{
        TextView itemName, itemPrice, itemType,itemQuantity;
        ImageView itemImage;
        Button btnRemove;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);

            itemImage = itemView.findViewById(R.id.Checkout_image);
            itemName = itemView.findViewById(R.id.checkout_name);
            itemPrice = itemView.findViewById(R.id.checkout_price);
            itemQuantity = itemView.findViewById(R.id.checkout_quantity);
            btnRemove = itemView.findViewById(R.id.btnRemove);
//            itemType = itemView.findViewById(R.id.it)


        }
    }
}

File: Kotlin_App_1/app/src/main/java/com/example/group1_project1/CartAdapter.kt
package com.example.group1_project1

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.*
import com.google.firebase.storage.FirebaseStorage
import com.google.firebase.storage.StorageReference


var totalPrice = 0;
var auth: FirebaseAuth? = null
class CartAdapter(private val cartList: ArrayList<CartItems>) : RecyclerView.Adapter<CartAdapter.MyViewHolder>() {

    private lateinit var mDatabase: DatabaseReference

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {

        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.cart_cardview,parent,false)
        return MyViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {

        val currentItem = cartList[position]
        holder.itemName.text = currentItem.itemName;
        holder.itemPrice.text = "$" + currentItem.itemPrice;
        holder.itemQuantity.text = "Quantity: " + currentItem.quantity;
        val storRef: StorageReference = FirebaseStorage.getInstance().getReferenceFromUrl(currentItem.itemImage.toString())
        Glide.with(holder.itemImage.context).load(storRef).into(holder.itemImage)

        //pass total amount to cart
        totalPrice += currentItem.totalPrice

        val intent = Intent("MyTotalAmount")
        intent.putExtra("totalAmount", totalPrice.toString())
//        LocalBroadcastManager.getInstance(context).sendBroadcast(intent)
//        final DatabaseReference itemRef = getRef(position);
//        final String myKey = itemRef.getKey();
//
//        holder.btnRemove.setOnClickListener(View.OnClickListener {
//            mDatabase.child(myKey).removeValue()
//        })
        holder.btnRemove.setOnClickListener {
            deleteItem(position)
//            val ref = FirebaseDatabase.getInstance().reference
//            val applesQuery = ref.child("Cart").orderByChild(auth!!.currentUser!!.uid).orderByChild("itemName")
//
//            applesQuery.addListenerForSingleValueEvent(object : ValueEventListener {
//                override fun onDataChange(dataSnapshot: DataSnapshot) {
//                    for (appleSnapshot in dataSnapshot.children) {
//                        appleSnapshot.ref.removeValue()
//                    }
//                }
//
//                override fun onCancelled(databaseError: DatabaseError) {
//                    Log.e(TAG, "onCancelled", databaseError.toException())
//                }
//            })


        }


    }

    override fun getItemCount(): Int {
        return cartList.size
    }

    fun deleteItem(index:Int){
        cartList.removeAt(index)
        notifyDataSetChanged()
 }

//    fun deleteItem(index:Int){
//        val key = cartList[index]
//        //cartList.removeAt(index)
//        //notifyDataSetChanged()
//
//        val mPostReference = FirebaseDatabase.getInstance().getReference("Cart").child(auth!!.currentUser!!.uid).child(key.toString());
//        mPostReference.removeValue();
//
//    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){

        val itemImage : ImageView = itemView.findViewById(R.id.Checkout_image);
        val itemName : TextView = itemView.findViewById(R.id.checkout_name);
        val itemPrice : TextView = itemView.findViewById(R.id.checkout_price);
        val itemQuantity : TextView = itemView.findViewById(R.id.checkout_quantity);
        val btnRemove : Button = itemView.findViewById(R.id.btnRemove);


    }
}
